OMCompiler/Compiler/Template/CodegenFMUCommon.tpl

```diff
@@ -711,8 +711,8 @@ case (BASEUNIT(mol = mol, cd = cd, m = m, s = s, A = A, K = K, kg = kg, factor =
   let A_Value = if not intEq(A, 0) then ' A="<% A %>"' else ""
   let K_Value = if not intEq(K, 0) then ' K="<% K %>"' else ""
   let kg_Value = if not intEq(kg, 0) then ' kg="<% kg %>"' else ""
-  let factor_Value = if not intEq(realInt(factor), 1) then ' factor="<% factor %>"' else ""
-  let offset_Value = if not intEq(realInt(offset), 0) then ' offset="<% offset %>"' else ""
+  let factor_Value = if not realEq(factor, 1.0) then ' factor="<% factor %>"' else ""
+  let offset_Value = if not realEq(offset, 0.0) then ' offset="<% offset %>"' else ""
```

Comment:
@adrpo Does this work in mingw? It seems that @arun3688 had experienced some problems.

Reply:
realEq is not working correctly in mingw, 
```
 <UnitDefinitions>
    <Unit name =\"kg2\">
      <BaseUnit  kg=\"2\" factor=\"1.000000000000001\"/>
    </Unit>
  </UnitDefinitions>
```
The factor valued should not be populated here as it has default value 1.0

Reply:
realEq is defined as

```
function realEq
  input Real x1;
  input Real x2;
  output Boolean b;
algorithm
  b := x1 == x2;
  annotation(__OpenModelica_EarlyInline = true, __OpenModelica_BuiltinPtr = true);
end realEq;
```

So 
`realEq(1.000000000000001, 1.0)` is `false`

I don't know how we never run into this so far 🤷‍♀️

https://floating-point-gui.de/.

Reply:
Well, that is not a problem with `realEq`. What else should it return? 😄

Reply:
I think it should probably have some tolerance like we do for simulation results. Or we might need another function that does that.

Reply:
In this case, having a factor of 1.000000000000001 isn't really acceptable. So we should probably round the factor to some precision and do the comparison then.

Reply:
Perhaps `if (abs(v - 1.0) < 1e-15)`

Reply:
Or any of the compare functions that check for almost equality

Reply:
You are a bit late 😄 Jenkins has already approved the changes